source/parser.hera

```diff
@@ -7679,14 +7691,83 @@ TypeElement
     if (!ws) return [ dots, space, type ]
     return [ ws, dots, space, trimFirstSpace(type) ]
 
-NestedTypeList
-  PushIndent NestedType*:types PopIndent ->
+NestedTypeElementList
+  PushIndent NestedTypeElement*:types PopIndent ->
     if (types.length)
       return types
     return $skip
 
-NestedType
-  Nested TypeElement ArrayElementDelimiter
+NestedTypeElement
+  Nested:indent TypeElementList:list ArrayElementDelimiter:delimiter ->
+    const {length} = list
+    if (!length) return $skip
+    // Prepend indent to first element and append delimiter to last element
+    return list.map((e, i) => {
+      if (i === 0) e = prepend(indent, e)
+      if (i === length - 1) e = append(e, delimiter)
+      return e
+    })
+
+# NOTE: Nested bulleted tuple starts with an indentation.
+NestedTypeBulletedTuple
+  ( InsertSpace InsertOpenBracket ):open PushIndent NestedTypeBullet*:content InsertCloseBracket:close PopIndent ->
+    if (!content.length) return $skip
+    content = content.flat() // combine bullets into one array
+
+    // Remove implicit comma from last element
+    const last = content[content.length - 1]
+    if (Array.isArray(last)) {
+      if (last.at(-1).implicit) last.pop()
```

Comment:
This mutation may cause caching issues with backtracking on some parses.

If we move it to a post processing handler then it won’t be an issue or we could make in non-mutative.